options/path.c

```diff
@@ -100,59 +102,70 @@ static const char *mp_get_platform_path(void *talloc_ctx,
 
     if (fallback_type) {
         mp_assert(strcmp(fallback_type, type) != 0);
-        return mp_get_platform_path(talloc_ctx, global, fallback_type);
+        return mp_get_platform_path(talloc_ctx, fallback_type);
     }
     return NULL;
 }
 
-void mp_init_paths(struct mpv_global *global, struct MPOpts *opts)
+static void global_configdir_destroy(void *p)
+{
+    global_configdir = NULL;
+}
+
+void mp_init_paths(struct MPOpts *opts)
 {
-    TA_FREEP(&global->configdir);
+    TA_FREEP(&global_configdir);
 
     const char *force_configdir = getenv("MPV_HOME");
     if (opts->force_configdir && opts->force_configdir[0])
         force_configdir = opts->force_configdir;
     if (!opts->load_config)
         force_configdir = "";
 
-    global->configdir = talloc_strdup(global, force_configdir);
+    global_configdir = talloc_strdup(opts, force_configdir);
+    talloc_set_destructor(global_configdir, global_configdir_destroy);
 }
 
-char *mp_find_user_file(void *talloc_ctx, struct mpv_global *global,
+char *mp_find_user_file(void *talloc_ctx, struct mp_log *log,
                         const char *type, const char *filename)
 {
     void *tmp = talloc_new(NULL);
-    char *res = (char *)mp_get_platform_path(tmp, global, type);
+    char *res = (char *)mp_get_platform_path(tmp, type);
     if (res)
         res = mp_path_join(talloc_ctx, res, filename);
     talloc_free(tmp);
-    MP_DBG(global, "%s path: '%s' -> '%s'\n", type, filename, res ? res : "-");
+    if (log)
+        mp_dbg(log, "%s path: '%s' -> '%s'\n", type, filename, res ? res : "-");
     return res;
 }
 
 static char **mp_find_all_config_files_limited(void *talloc_ctx,
-                                               struct mpv_global *global,
+                                               struct mp_log *log,
                                                int max_files,
                                                const char *filename)
 {
     char **ret = talloc_array(talloc_ctx, char*, 2); // 2 preallocated
     int num_ret = 0;
 
     for (int i = 0; i < MP_ARRAY_SIZE(config_dirs); i++) {
-        const char *dir = mp_get_platform_path(ret, global, config_dirs[i]);
+        const char *dir = mp_get_platform_path(ret, config_dirs[i]);
         bstr s = bstr0(filename);
         while (dir && num_ret < max_files && s.len) {
             bstr fn;
             bstr_split_tok(s, "|", &fn, &s);
 
             char *file = mp_path_join_bstr(ret, bstr0(dir), fn);
             if (mp_path_exists(file)) {
-                MP_DBG(global, "config path: '%.*s' -> '%s'\n",
-                        BSTR_P(fn), file);
+                if (log) {
```

Comment:
This makes this log useless, because we will not see it in "some" code paths.

Reply:
What do you mean? It should be whatever log is passed down sans the NULL case but nothing uses that.

Reply:
What's the purpose of this check is nothing uses that?

Reply:
I just thought to make the log optional for the helpers in case some use in the future would want to use this without needing the pass an mp_log.